doc/usage/extensions/napoleon.rst

```diff
@@ -496,3 +535,36 @@ sure that "sphinx.ext.napoleon" is enabled in `conf.py`::
    If an entry is just a string, it is interpreted as a header for a generic
    section. If the entry is a tuple/list/indexed container, the first entry
    is the name of the section, the second is the section key to emulate.
+
+.. confval:: napoleon_numpy_returns_no_rtype
+
+   *Defaults to False.*
+   Traditional `NumPy style`_ requires the return type and a description
+   provided indented underneath::
+
+       Returns
+       -------
+       bool
+           A description of the return.
+       int
+           A different description for a different possible return.
+
+   This style affords the ability to document multiple return types, but
+   the formatting rules inherently prohibit omitting the return type.  If
+   using :confval:`autodoc_typehints` as ``'description'``, and setting
+   this configuration value to ``True``, a user may omit the return type:
+
+   .. code-block:: diff
+
+        Returns
+        -------
+      + A description of the return.
+      - bool
+      -     A description of the return.
+
+   As python function signatures may only have one return type annotation,
+   setting this value to True implies only one return type is being
+   documented.
```

Comment:
Is it necessary to clarify with an example of returning with `Union[int, str]` or is this clear enough?  It's already kind of long-winded...

Reply:
Indeed. It would be better to do that, but it is a bit difficult...